packages/shared/src/utils/spec.ts

```diff
@@ -31,37 +22,80 @@ export function StringUnion<T extends string[]>(values: [...T], options?: any) {
   );
 }
 
-export function parseTypeBox(spec: TSchema, noOptional = false): Static<typeof spec> {
-  if (spec.modifier === OptionalModifier && !noOptional) {
-    return undefined;
+function getArray(items: JSONSchema7Definition[]): JSONSchema7Type[] {
+  return items.map(item => (isJSONSchema(item) ? parseTypeBox(item) : null));
+}
+
+function getObject(spec: JSONSchema7): JSONSchema7Object {
+  const obj: JSONSchema7Object = {};
+
+  if (spec.allOf && spec.allOf.length > 0) {
+    return (getArray(spec.allOf) as JSONSchema7Object[]).reduce((prev, cur) => {
+      prev = Object.assign(prev, cur);
+      return prev;
+    }, obj);
+  }
+
+  for (const key in spec.properties) {
+    const subSpec = spec.properties[key];
+    if (typeof subSpec === 'boolean') {
+      obj[key] = null;
+    } else {
+      obj[key] = parseTypeBox(subSpec);
+    }
+  }
+  return obj;
+}
+
+export function parseTypeBox(spec: JSONSchema7): JSONSchema7Type {
+  // compatible with typebox optional
+  if ((spec as TSchema).modifier === OptionalModifier) {
```

Comment:
May check the optional property by object type schema's `required` properties in the `getObject` function instead of using the `typebox`'s `OptionalModifier`?

Reply:
Good idea, I'll change it

Reply:
done